### PR TITLE
Refactor push, pull commands

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
Refactor push, pull commands

### CODE REVIEW

File: components/cli/pkg/commands/build_test.go [hunk 1]
```diff
@@ -134,8 +134,10 @@ func TestExecuteTempBalFile(t *testing.T) {
 }
 
 func TestGenerateMetaData(t *testing.T) {
-	mockCli := test.NewMockCli(nil)
-	projectDir, _ := mockCli.FileSystem().CurrentDir()
+	currentDir, err := ioutil.TempDir("", "current-dir")
```
Comment (isurulucky) [inline components/cli/pkg/commands/build_test.go]: Check for the error.

File: components/cli/pkg/commands/list_ingresses.go [hunk 2]
```diff
@@ -202,14 +204,14 @@ func displayImageApisTable(imageName string) {
...
+		displayCompositeImageApisTable(cli, imageName)
 	}
 }
 
-func displayCompositeImageApisTable(compositeImageContent string) {
-	cell, err := getIngressValues(compositeImageContent)
+func displayCompositeImageApisTable(cli cli.Cli, compositeImageContent string) {
+	cell, err := getIngressValues(cli, compositeImageContent)
 	if err != nil {
 		util.ExitWithErrorMessage("Error occurred while displaying composite image ingress", err)
```
Comment (isurulucky) [inline components/cli/pkg/commands/list_ingresses.go]: Lets return this error and handle from parent level. Please fix in all places.

File: components/cli/pkg/registry/registry.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+		parsedCellImage.ImageVersion)
+	repository := parsedCellImage.Organization + "/" + parsedCellImage.ImageName
+	// Creating the Cell Image Digest (Docker file Layer digest)
+	hash := sha256.New()
+	hash.Write(fileBytes)
+	sha256sum := hex.EncodeToString(hash.Sum(nil))
+	cellImageDigest := digest.NewDigestFromEncoded(digest.SHA256, sha256sum)
+
+	// Checking if the the Cell Image already exists in the registry
+	fmt.Println(fmt.Sprintf("\nChecking if the image %s already exists in the Registry", util.Bold(imageName)))
```
Comment (isurulucky) [inline components/cli/pkg/registry/registry.go]: Use the stdout provided by Out() function in registry interface to print.

File: components/cli/pkg/registry/registry.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+	// Pushing the cell image if it is not already uploaded
+	if !cellImageDigestExists {
+		fmt.Println(fmt.Sprintf("Pushing image %s", util.Bold(imageName)))
+		// Read stream of files
+		err = hub.UploadBlob(repository, cellImageDigest, bytes.NewReader(fileBytes), nil)
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println(fmt.Sprintf("\nUsing already existing image %s in %s Registry", util.Bold(imageName),
```
Comment (isurulucky) [inline components/cli/pkg/registry/registry.go]: Use the stdout provided by Out() function in registry interface to print.

File: components/cli/pkg/registry/registry.go [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+	if err != nil {
+		return fmt.Errorf("error occurred while pushing the cell image, %v", err)
+	}
+
+	// Uploading the manifest to the Cellery Registry (Docker Registry)
+	err = hub.PutManifest(repository, parsedCellImage.ImageVersion, signedCellImageManifest)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("\nImage Digest : %s\n", util.Bold(cellImageDigest))
```
Comment (isurulucky) [inline components/cli/pkg/registry/registry.go]: Use the stdout provided by Out() function in registry interface to print.

File: components/cli/pkg/commands/build_test.go (hunk 1 above)
Comment (MadhukaHarith92) [reply components/cli/pkg/commands/build_test.go]: Updated.

File: components/cli/pkg/commands/list_ingresses.go (hunk 2 above)
Comment (MadhukaHarith92) [reply components/cli/pkg/commands/list_ingresses.go]: list ingresses command is not refactored yet. Will updated and return the error in the next PR.

File: components/cli/pkg/registry/registry.go (hunk 3 above)
Comment (MadhukaHarith92) [reply components/cli/pkg/registry/registry.go]: Updated.

File: components/cli/pkg/registry/registry.go (hunk 4 above)
Comment (MadhukaHarith92) [reply components/cli/pkg/registry/registry.go]: Updated.

File: components/cli/pkg/registry/registry.go (hunk 5 above)
Comment (MadhukaHarith92) [reply components/cli/pkg/registry/registry.go]: Updated.

**Pull request merged**